.github/workflows/build-and-push.yml

```
@@ -0,0 +1,70 @@
+name: Build, test and push container image
+on:
+  workflow_call:
+    inputs:
+      continue-on-error:
+        type: string
+      build_args:
+        type: string
+      cmake_args:
+        type: string
+      dockerfile:
+        type: string
+      tag:
+        type: string
+
+env:
+  docker-tag: ghcr.io/kokkos/ci-containers/${{ inputs.dockerfile }}:${{ inputs.tag }}
+
+jobs:
+  build-test-push:
+    continue-on-error: ${{ inputs.continue-on-error == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v2
+        if: github.repository_owner == 'kokkos' && ( github.event_name == 'push' ||  github.event_name == 'schedule' )
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Convert build_args
+        id: build_args
+        run: |
+          echo "args<<EOF" >> $GITHUB_OUTPUT
+          echo ${{ inputs.build_args }} | sed 's/,/\n/' >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+      - name: Build new image and Load it into Docker for testing
+        uses: docker/build-push-action@v4
+        with:
+          tags: ${{ env.docker-tag }}
+          file: ${{ inputs.dockerfile }}
+          build-args: ${{ steps.build_args.outputs.args }}
+          pull: true
+          load: true
+      - name: Checkout kokkos
+        uses: actions/checkout@v3
+        with:
+          repository: kokkos/kokkos
+          path: kokkos
+      - name: Build and Test Kokkos using updated image
+        run: |
+          docker run -t --rm -v ${{ github.workspace }}/kokkos:/kokkos -w /kokkos ${{ env.docker-tag }} \
+            sh -c \
+              "cmake -B builddir \
+                ${{ inputs.cmake_args }} \
+                -DKokkos_ENABLE_HWLOC=ON \
+                -DKokkos_ENABLE_BENCHMARKS=ON \
+                -DKokkos_ENABLE_EXAMPLES=ON \
+                -DKokkos_ENABLE_TESTS=ON && \
```

I think with
```suggestion
                -DKokkos_ENABLE_HWLOC=ON \
                -DKokkos_ENABLE_BENCHMARKS=ON \
                -DKokkos_ENABLE_EXAMPLES=ON \
                -DKokkos_ENABLE_TESTS=ON && \
                -DKokkos_ENABLE_SERIAL=ON && \
                -DKokkos_ENABLE_OPENMP=ON && \              
```
we would have a pretty good coverage of what we are doing in the Kokkos GitHub CI.

---

Done.